includes/REST_API/Stories_Media_Controller.php

```diff
@@ -128,17 +134,43 @@ public function get_items( $request ) {
 	public function create_item( $request ) {
 		// WP_REST_Attachments_Controller doesn't allow setting an attachment as the parent post.
 		// Hence we are working around this here.
-		$parent_post = ! empty( $request['post'] ) ? (int) $request['post'] : null;
-		$original_id = ! empty( $request['original_id'] ) ? (int) $request['original_id'] : null;
+
+		/**
+		 * Parent post.
+		 *
+		 * @var int $parent_post
+		 */
+		$parent_post = ! empty( $request['post'] ) ? $request['post'] : null;
+
+		/**
+		 * Original post ID.
+		 *
+		 * @var int $original_id
+		 */
+		$original_id = ! empty( $request['original_id'] ) ? $request['original_id'] : null;
+
 		unset( $request['post'] );
 
 		$response = parent::create_item( $request );
-		if ( is_wp_error( $response ) || ( ! $parent_post && ! $original_id ) ) {
+		if ( ( ! $parent_post && ! $original_id ) || is_wp_error( $response ) ) {
```

Comment:
Why does this have to change?

Reply:
Microperformance, checking two variables is faster than a function call.